test/test-parse-warning.cpp

```diff
@@ -169,6 +169,164 @@ TEST_F(test_error_equals_does_not_distribute_over_or, non_constant) {
     EXPECT_THAT(p.errors, IsEmpty());
   }
 }
+
+TEST_F(test_parse_warning, warn_on_pointless_string_compare) {
+  {
+    test_parser p(u8"s.toLowerCase() == 'banana'"_sv, capture_diags);
+    p.parse_and_visit_statement();
+    EXPECT_THAT(p.errors, IsEmpty());
```

Comment:
Nit: No need to capture diags if we are asserting no errors.
```suggestion
    test_parser p(u8"s.toLowerCase() == 'banana'"_sv);
    p.parse_and_visit_statement();
```